src/authentication/AuthenticationCallbackPage.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect } from 'react';
+import { Spinner } from 'theme-ui';
+import { useHistory, useLocation } from 'react-router-dom';
+import http from 'axios';
+import { authenticate, AuthenticationMethod } from '../api';
+import { useAuthentication } from './context';
+import { decode } from './token';
+
+export const AuthenticationCallbackPage = () => {
+  const queryParams = useQuery();
+  const { authenticate: saveToken } = useAuthentication();
+  const history = useHistory();
+
+  useEffect(() => {
+    const cancelToken = http.CancelToken.source();
+
+    async function authenticateWithMethod() {
+      const method = AuthenticationMethod[queryParams.get('method') as AuthenticationMethod];
+      const authCode = queryParams.get(authenticationMethodAuthCodeQueryParameter[method]);
+      if (method && authCode) {
```

Comment:
Should there be an else on this `if`?

Reply:
Yup, I guess we could also `history.replace(`/login?invalid=true`);`?

Reply:
Actually I can simplify it by just removing the check. If we fail, it will redirect anyhow.